### ISSUE TITLE
For complete back up options add Contact, sms, and wifi

### ISSUE THREAD

**Issue by rancidfrog** [machiav3lli/oandbackupx#86]
Having the ability to add sms and wifi to backup is a great feature.

There are open source solution to backup sms:
https://github.com/hydrargyrum/epistolaire

For restore, maybe stack has solutions, i.e.
https://stackoverflow.com/questions/10866001/how-to-restore-the-sms-backup-taken-to-sms-db-in-android

Wifi there must be checks for which version is used,
older android version stores data at /data/misc/wifi/wpa_supplicant.conf

For contacts there should be .vcf import export or similar solution

**Comment by DamienCassou** [comment]
I agree with that: It would be great to get text messages in the backup.

**Comment by kocjs** [comment]
I am looking for this too, especially automatized .vcf import export. 

A valuable future, this would make the backup re-storable by default. I a worst case scenario contacts might re-storable only to the exact same device with the exact same software on it.

**Comment by euphoria360** [comment]
me too. This would greatly improve usecase.
Some alternatives (like Swift Backup. although its a premium app) have this.

BTW why in the hell these 2 apps look so familiar? does that mean Swift Backup forked OandBackup too?

**Comment by Tiefkuehlpizze** [comment]
Hi, I understand your usecases and I would also like to see this working.

Regarding Swift Backup similarities: 
I don't think, they are related. It might be coincidence, that the style looks a little bit similar. It's maybe just inspiration. Afaik Oandbackupx did not copy anything from Swift Backup (their code is closed source and obfuscated) and I don't think, that OABX wants to be a clone of SwiftBackup. 
It's going its own way and I can promise, that it will never force users to provide a Google/any other Cloud Account. It would be against F-Droids rules to include proprietary libraries and personally this is the reason why I prefer OABX.

**Comment by machiav3lli** [comment]
OAndBackup had (for the most part) a whole different UI than both. I never used Swift Backup but from the fact that it needs GSF I could guess that's based on a whole other backend and backup logic using FSF for functionality

**Comment by machiav3lli** [comment]
And I add my voice to @Tiefkuehlpizze that GSF isn't a viable option for OABX development, personally and based on FOSS principles.

**Comment by euphoria360** [comment]
> Hi, I understand your usecases and I would also like to see this working.
> 
> Regarding Swift Backup similarities:
> I don't think, they are related. It might be coincidence, that the style looks a little bit similar. It's maybe just inspiration. Afaik Oandbackupx did not copy anything from Swift Backup (their code is closed source and obfuscated) and I don't think, that OABX wants to be a clone of SwiftBackup.
> It's going its own way and I can promise, that it will never force users to provide a Google/any other Cloud Account. It would be against F-Droids rules to include proprietary libraries and personally this is the reason why I prefer OABX.

Thanks for the quick reply.
Yes, SwiftBackup depends heavily on GSM. they don't even allow using the app if one doesn't own a Google account.

I have bought their subscription, but the fact of them forcing me to own a Google account is driving me away.

Thank god you exist.

**Comment by Tiefkuehlpizze** [comment]
Just a quick update on this issue: While working on the app, I figured out, that you can simply backup the data of "Phone and Messaging Storage" (com.android.providers.telephony). For contacts you might be luck with "Contacts Storage" (com.android.providers.contacts).

Just restore the data and reboot afterwards to make your changes visible.

**Comment by hagaygo** [comment]
This might work if you backup/restore excatly on the same version of android and flavor (AOSP/OEM).

I almost sure this idea would not work if you are moving to other android version and maybe totally other phone.

Other apps uses API as "default SMS app" to restore/import messages and hence works fine without reboot (and between roms).

**Comment by opusforlife2** [comment]
There's also the very old but surprisingly still working Slight Backup (call logs and messages work just fine): 

Slight backup (A very simple backup program) - https://f-droid.org/packages/de.shandschuh.slightbackup

Don't know how useful this will be, but here's the repo: https://github.com/handschuh/Slight-backup

**Comment by dl200010** [comment]
Like hagaygo said there are other backup solutions that use "default SMS app" to backup. I know Titanium Backup does this. It is very out of date and now I have to switch it to default before restoring. It used to do it on its own.

**Comment by machiav3lli** [comment]
@dl200010 you seem to be already working on this. You can gladly ask questions and report progress here.

**Comment by dl200010** [comment]
Yeah, it will be slow going because I am learning as I go. I did learn Java in college (BS in ComSci), which is helping with Kotlin. If you have any suggestions from my commits, then please say something. That way I can keep consistent with the code. Also, please verify I am doing the license part right. I will pull request when I am happy with my changes.

Currently I do have OABX appearing as a default SMS app and saving incoming SMS to the phone database (no notifications, yet). I noticed that when set to default SMS incoming SMS would be lost. I have to figure out the MMS side of that. I am currently only working on SMS/MMS. My plan is to read the local SMS database and save to JSON and restore back to the local SMS database. Once that is done I may go on to do other parts, like WiFi APs and Contacts. Contacts isn't a high priority for me, personally, because I use GMail to sync mine.


https://github.com/dl200010/oandbackupx <<< For anyone that wants to look.

**Comment by machiav3lli** [comment]
Nice and thanks, Kotlin makes things easier and shorter. But be careful to not fall in love with it.

Cool that's something that can go in the next stable release. For a more export/import type of backup you can look up the code at https://github.com/tmo1/sms-ie it can be a source of inspiration (kudos to the developer if used would be nice).

**Comment by dl200010** [comment]
It is what it is. I don't fall in love with any language. I just learn what to use them for and find another if the ones I know don't work.

May not be in the next release, or two. Depending on how long it takes me. I am using https://github.com/tmo1/sms-ie and a few other code bases as inspiration only. I want to learn what the code I am writing is doing. I don't want to copy and paste something in and it works, but not know how or why.

**Comment by dl200010** [comment]
SMS/MMS backups are on their way. https://github.com/machiav3lli/oandbackupx/pull/477

**Comment by dl200010** [comment]
https://github.com/machiav3lli/oandbackupx/pull/479

**Comment by dl200010** [comment]
https://github.com/machiav3lli/oandbackupx/pull/481

**Comment by 07416** [comment]
To anyone who isn't aware, [QKSMS](https://github.com/moezbhatti/qksms) can import & export SMS messages without root. 

I've understood that you can change your default SMS app after the process has been completed, if you so desire. 

At least the LineageOS contacts app `com.android.contacts` supports import & export (.vcf file).

**Comment by dl200010** [comment]
> To anyone who isn't aware, [QKSMS](https://github.com/moezbhatti/qksms) can import & export SMS messages without root. I've understood that you can change your default SMS app after the process has been completed, if you so desire.
> 
> At least the LineageOS contacts app `com.android.contacts` supports import & export (.vcf file).

So can Neo Backup, but root is required because other parts of the app requires it. Neo Backup never issues any root command to backup SMS/MMS or Call Logs.
